canonical-web-and-design/ubuntu.com
# WIP Cube purchase verification

## Done

- Cube microcerts purchase verification endpoint


## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/cube/microcerts

This is the 4th step of the complete purchase flow:
1. User gets list of courses `GET /cube/microcerts.json?test_backend=true`
2. User preview-purchases a course/exam `POST /cube/microcerts/purchase.json?test_backend=true&preview=true`
3. User actual purchase the same course/exam `POST /cube/microcerts/purchase.json?test_backend=true` (same endpoint no preview param)
4. *Immediately after call this endpoint using id given by previous endpoint* `GET /cube/microcerts/purchases/<purchase_id>`

Therefore, to QA this PR, you only need to pass the corresponding `purchase_id` given in endpoint 3.

Response structure:
```
{
  accountID: "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
  createdAt: "2021-10-05T09:29:22.134Z"
  createdBy: "XxXxXX"
  id: "ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ" // purchase id
  invoice: {amountDue: 1200, amountPaid: 1200, currency: "usd",…}
  lastModified: "2021-10-05T09:29:22.134Z"
  marketplace: "canonical-cube"
  purchaseItems: [
     {
       productListingID: "ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ", // productid required by endpoint 2
       value: 1
     }
  ]
  status: "processing"
  stripeInvoices: [
     {id: "in_xXxxxXXxxxxXXXXX", last_update: "2021-10-05T09:29:26.02357Z", pi_status: "requires_payment_method"}
  ]
```
  
## Issue / Card

Fixes https://github.com/canonical-web-and-design/commercial-squad/issues/260
